src/abstractions/serialization/SerializationWriterFactoryRegistry.cs

```diff
@@ -51,5 +51,45 @@ public ISerializationWriter GetSerializationWriter(string contentType)
             throw new InvalidOperationException($"Content type {cleanedContentType} does not have a factory registered to be parsed");
         }
 
+        /// <summary>
+        /// Get the relevant <see cref="ISerializationWriter"/> instance for the given content type
+        /// </summary>
+        /// <param name="contentType">The content type in use</param>
+        /// <param name="serializeOnlyChangedValues">If <see langword="true"/> will only return changed values, otherwise will return the full object </param>
+        /// <returns>A <see cref="ISerializationWriter"/> instance to parse the content</returns>
+        public ISerializationWriter GetSerializationWriter(string contentType, bool serializeOnlyChangedValues)
+        {
+            if(serializeOnlyChangedValues)
+                return GetSerializationWriter(contentType);
+
+            var factory = GetSerializationWriterFactory(contentType);
+            if(factory is Store.BackingStoreSerializationWriterProxyFactory backingStoreFactory)
+                return backingStoreFactory.GetSerializationWriter(contentType, false);
+
+            return factory.GetSerializationWriter(contentType);
+        }
+
+        /// <summary>
+        /// Get the relevant <see cref="ISerializationWriterFactory"/> instance for the given content type
+        /// </summary>
+        /// <param name="contentType">The content type in use</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        private ISerializationWriterFactory GetSerializationWriterFactory(string contentType)
```

Comment:
I believe the GetSerializationWriter could use this method and return the writer instead of duplicating the code.

Reply:
Good point